app/src/main/java/ch/epfl/skysync/database/tables/AvailabilityTable.kt

```diff
@@ -48,6 +63,45 @@ class AvailabilityTable(db: FirestoreDatabase) :
       db.setItem(path, availabilityId, AvailabilitySchema.fromModel(userId, item))
     }
   }
+  /**
+   * Returns the available user on the given day and timeslot
+   *
+   * @param localDate The requested day
+   * @param timeslot The requested timeslot
+   */
+  suspend fun getUsersAvailableOn(localDate: LocalDate, timeslot: TimeSlot): List<User> =
+      coroutineScope {
+        val dateFilter = Filter.equalTo("date", DateLocalDateConverter.localDateToDate(localDate))
+        val timeslotFilter = Filter.equalTo("timeSlot", timeslot)
+        val dateTimeSlotFilter = Filter.and(dateFilter, timeslotFilter)
+
+        // Retrieve all flights of the given day, timeslot
+        val flightsIds = flightTable.query(dateTimeSlotFilter).map { flight: Flight -> flight.id }
+        // Retrieve all members of these flights (they are not available)
+        val unavailableUserIds =
+            flightMemberTable.query(Filter.arrayContainsAny("flightId", flightsIds)).map { fm ->
```

Comment:
`arrayContainsAny` has limitations (see [here](https://firebase.google.com/docs/firestore/query-data/queries?_gl=1*7wgdrw*_up*MQ..*_ga*MTQwMzAzMzQyNi4xNzE0Mzg3MzA0*_ga_CW55HF8NVT*MTcxNDM4NzMwMy4xLjAuMTcxNDM4NzMwMy4wLjAuMA..#array-contains-any)) that you want to take into considerations, here it should be fine as I doubt they is going to be more than 30 flights on a given time slots. Just so you know

Reply:
I used the wrong function. The right one should be inArray, which also has the same limitation. Here, I won't change it because I assume that the number of flights for a given date and time slot is less than 30.